src/apps/openfluid-devstudio/WareBuildStatusWidget.cpp

```diff
@@ -71,27 +97,109 @@ WareBuildStatusWidget::~WareBuildStatusWidget()
 // =====================================================================
 
 
-openfluid::ui::common::ActionLabel* WareBuildStatusWidget::installLabel()
+void WareBuildStatusWidget::refreshBuildJobsStatus()
+{
+  QString JobStatusString;
+  if (m_BuildOptions.IsMultipleJobs)
+  {
+    JobStatusString = tr("%1 jobs").arg(m_BuildOptions.JobsNumber);
+  }
+  else
+  {
+    JobStatusString = tr("single job");
+  }
+  ui->jobStatus->setText(JobStatusString);
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareBuildStatusWidget::refreshBuildModeStatus()
 {
-  return ui->buildStatus;
+  if (m_BuildOptions.IsWithInstall)
+  {
+    ui->buildStatus->setText(tr("install"));
+  }
+  else
+  {
+    ui->buildStatus->setText(tr("no install"));
+  }
 }
 
 
 // =====================================================================
 // =====================================================================
 
 
-openfluid::ui::common::ActionLabel* WareBuildStatusWidget::configureModeLabel()
+void WareBuildStatusWidget::refreshConfigureModeStatus()
 {
-  return ui->configStatus;
+  if (m_BuildOptions.IsReleaseMode)
+  {
+    ui->configStatus->setText(tr("release"));
+  }
+  else
+  {
+    ui->configStatus->setText(tr("debug"));
+  }
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareBuildStatusWidget::onJobsClicked()
+{
+  m_BuildOptions.IsMultipleJobs = !m_BuildOptions.IsMultipleJobs;
+  emit settingsChanged(m_BuildOptions);
+  refreshBuildJobsStatus();
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareBuildStatusWidget::onJobsScrolled(bool Up)
+{
+  if (m_BuildOptions.IsMultipleJobs)
+  {
+    if (Up)
+    {
+      m_BuildOptions.JobsNumber++;
+    }
+    else if (m_BuildOptions.JobsNumber > 2)
+    {
+      m_BuildOptions.JobsNumber--;
+    }
+    emit settingsChanged(m_BuildOptions);
```

Comment:
as Qt signals are blocking, emit them at the end, as the last instruction of task